File: frontend/src/services/trackingService.js
import api from './api';

export const trackingService = {
  getTimeline: async (trackingNumber) => {
    const response = await api.get(`/api/v1/tracking/${trackingNumber}`);
    return response.data;
  },
  getLatestStatus: async (trackingNumber) => {
    const response = await api.get(`/api/v1/tracking/${trackingNumber}/latest`);
    return response.data;
  },
  updateStatus: async (trackingNumber, payload) => {
    // payload: { shipmentStatus, location, remarks }
    const response = await api.post(`/api/v1/tracking/${trackingNumber}`, payload);
    return response.data;
  },
};

File: deliverytracker/src/main/java/com/lastmile/deliverytracker/tracking/service/TrackingService.java
package com.lastmile.deliverytracker.tracking.service;

import com.lastmile.deliverytracker.tracking.dto.request.UpdateTrackingRequest;
import com.lastmile.deliverytracker.tracking.dto.response.TrackingHistoryResponse;
import com.lastmile.deliverytracker.tracking.dto.response.TrackingTimelineResponse;

public interface TrackingService {

    TrackingTimelineResponse getTimeline(String trackingNumber);

    TrackingHistoryResponse updateTracking(String trackingNumber, UpdateTrackingRequest request);

    TrackingHistoryResponse getLatestStatus(String trackingNumber);
}
